pkg/service/repair/controller.go

```diff
@@ -177,50 +172,39 @@ func (c *rowLevelRepairController) TryBlock(hosts []string) (bool, allowance) {
 	return true, a
 }
 
+func (c *rowLevelRepairController) parallelLimit() int {
+	p := c.intensity.Parallel()
+
+	if p <= defaultParallel {
+		return c.intensity.MaxParallel()
+	}
+
+	return p
+}
+
 func (c *rowLevelRepairController) shouldBlock(hosts []string, intensity float64) bool {
 	// ALLOW if nothing is running
 	if !c.Busy() {
 		return true
 	}
 
-	// DENY if any host has max nr of jobs already running
+	// DENY if any host has a job running already. We don't allow more than a single repair job per host.
 	for _, h := range hosts {
-		if c.jobs[h] >= c.limits[h].Default {
+		if c.jobs[h] >= 1 {
 			return false
 		}
 	}
 
-	// DENY if any host already runs at the full intensity.
-	// This may happen if we decrease intensity, for a while we need to reduce
-	// number of jobs to reduce number of busy ranges.
-	ranges := c.rangesForIntensity(hosts, intensity)
-	for _, h := range hosts {
-		var (
-			r  = c.busyRanges[h] + ranges
-			ok bool
-		)
-		switch {
-		case intensity == maxIntensity:
-			ok = r <= c.limits[h].Max
-		case intensity <= 1:
-			ok = r <= c.limits[h].Default
-		default:
-			ok = r <= int(intensity)*c.limits[h].Default && r <= c.limits[h].Max
-		}
-		if !ok {
-			return false
-		}
+	// DENY (Sanity) if allJobs + 1 > MaxParallel.
+	// However, this should never happen due to "up to 1 repair task per replica" and the fact that MaxParallel is
+	// calculated the way that allows up to 1 repair task per host.
+	if c.allJobs+1 > c.intensity.MaxParallel() {
```

Comment:
I assume c.allJobs tracks the number of pending jobs that are currently running. 

nit: We can use >= to avoid the plus one thing. Saying something in the comments like, we already have c.allJobs running and we can not have more than MaxParallel() jobs, so we need to block the request. 
```
if c.allJobs >= c.intensity.MaxParallel()  {
}
```

Reply:
Correct the result would be the same.
However I followed the previous code approach on this and wanted to keep different condition blocks (some of when remained) to look similar.

And hence the +1 thingy.
In practice, this is not a real-time application anyway so it's not about the performance to begin with.

And I kinda think that `if current_counter+1> max_counter; then {don't do something} else {++current_counter;}` is more readable than `>=` option - it semantically hints that we are checking a condition that if held is going to cause the `current_counter` to be incremented - thereby self-explaining its (`current_counter`) semantics.

Reply:
No, this does not bother me much and I was not mentioning this for any perf reasons. I just want to make sure I understand the code correctly (the plus one thing).